### ISSUE TITLE
[RyuJIT/arm32] Assertion failed 'recentAssignedRef->delayRegFree'

### ISSUE THREAD

**Issue by BruceForstall** [dotnet/coreclr#14546]
From corefx x86_arm_altjit run, e.g. "@dotnet-bot test Windows_NT x86_arm_altjit Checked corefx_baseline".

Found in System.Numerics.Vectors.Tests:
```
E:\gh\corefx\bin\tests\System.Numerics.Vectors.Tests\netcoreapp-Windows_NT-Release-x86>RunTests.cmd E:\gh\corefx\bin\testhost\netcoreapp-Windows_NT-Release-x86\
Using E:\gh\corefx\bin\testhost\netcoreapp-Windows_NT-Release-x86\ as the test runtime folder.
Executing in E:\gh\corefx\bin\tests\System.Numerics.Vectors.Tests\netcoreapp-Windows_NT-Release-x86\
Running tests... Start time: 13:31:45.85

E:\gh\corefx\bin\tests\System.Numerics.Vectors.Tests\netcoreapp-Windows_NT-Release-x86>set COMPlus_AltJit=*
E:\gh\corefx\bin\tests\System.Numerics.Vectors.Tests\netcoreapp-Windows_NT-Release-x86>set COMPlus_AltJitNgen=*
E:\gh\corefx\bin\tests\System.Numerics.Vectors.Tests\netcoreapp-Windows_NT-Release-x86>set COMPlus_AltJitName=protononjit.dll
E:\gh\corefx\bin\tests\System.Numerics.Vectors.Tests\netcoreapp-Windows_NT-Release-x86>set COMPlus_NoGuiOnAssert=1
E:\gh\corefx\bin\tests\System.Numerics.Vectors.Tests\netcoreapp-Windows_NT-Release-x86>set COMPlus_ContinueOnAssert=0
E:\gh\corefx\bin\tests\System.Numerics.Vectors.Tests\netcoreapp-Windows_NT-Release-x86>set COMPlus_AltJitAssertOnNYI=1

E:\gh\corefx\bin\tests\System.Numerics.Vectors.Tests\netcoreapp-Windows_NT-Release-x86>call E:\gh\corefx\bin\testhost\netcoreapp-Windows_NT-Release-x86\\dotnet.exe xunit.console.netcore.exe System.Numerics.Vectors.Tests.dll  -xml testResults.xml -notrait Benchmark=true -notrait category=nonnetcoreapptests -notrait category=nonwindowstests  -notrait category=IgnoreForCI -notrait category=OuterLoop -notrait category=failing
xUnit.net console test runner (32-bit .NET Core)
Copyright (C) 2014 Outercurve Foundation.

Discovering: System.Numerics.Vectors.Tests
Discovered:  System.Numerics.Vectors.Tests
Starting:    System.Numerics.Vectors.Tests

Assert failure(PID 688 [0x000002b0], Thread: 17816 [0x4598]): Assertion failed 'recentAssignedRef->delayRegFree' in 'System.Numerics.Tests.GenericVectorTests:IsDiffTolerable(float,float,int):bool' (IL size 76)

    File: e:\gh\coreclr2\src\jit\lsra.cpp Line: 5942
    Image: E:\gh\corefx\bin\testhost\netcoreapp-Windows_NT-Release-x86\dotnet.exe
```

Full repro:
```
dotnet/coreclr:
E:\gh\coreclr2> build.cmd x86 checked

dotnet/corefx:
E:\gh\corefx> build.cmd -Release -os:Windows_NT -buildArch:x86 -- /p:CoreCLROverridePath=E:\gh\coreclr2\bin\Product\Windows_NT.x86.Checked
E:\gh\corefx> build-tests.cmd -Release -os:Windows_NT -buildArch:x86 -- /p:WithoutCategories=IgnoreForCI /p:PreExecutionTestScript=e:\bugs\SetStressModes.bat

With e:\bugs\SetStressModes.bat:
set COMPlus_AltJit=*
set COMPlus_AltJitNgen=*
set COMPlus_AltJitName=protononjit.dll
set COMPlus_NoGuiOnAssert=1
set COMPlus_ContinueOnAssert=0
set COMPlus_AltJitAssertOnNYI=1
```

**Comment by BruceForstall** [comment]
@dotnet/arm32-contrib @CarolEidt

**Comment by YongseopKim** [comment]
Different assert failure on linux/arm32

```
 dragon@dragon-rasp3  ~/dotnet/arm32-dotnet  ./run_fxutc_ryujit_arm.sh 
LOG: FX_RYUJIT_10241540.log
export COMPlus_AltJit="*"
export COMPlus_AltJitNgen="*"
export COMPlus_ContinueOnAssert=0
export COMPlus_AltJitAssertOnNYI=1

/home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10241504/run-test.sh --runtime /home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10241504/overlay --corefx-tests /home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10241504/tests --os Linux --arch x64 --configurationGroup Release --test-dir-file ./tests_fx2.txt

Running tests in /home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10241504/tests/System.Numerics.Vectors.Tests/netcoreapp-Linux-Release-x64 ./RunTests.sh /home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10241504/overlay

Using as the test runtime folder.
Running tests... Start time: 15:40:02
Commands:
export XUNIT_PERFORMANCE_MIN_ITERATION=1
export XUNIT_PERFORMANCE_MAX_ITERATION=1
chmod +x /home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10241504/overlay/dotnet
/home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10241504/overlay/dotnet xunit.console.netcore.exe System.Numerics.Vectors.Tests.dll  -xml testResults.xml -notrait category=nonnetcoreapptests -notrait category=nonlinuxtests  -notrait category=failing
~/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10241504/tests/System.Numerics.Vectors.Tests/netcoreapp-Linux-Release-x64 ~/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10241504/tests/System.Numerics.Vectors.Tests/netcoreapp-Linux-Release-x64
xUnit.net console test runner (32-bit .NET Core)
Copyright (C) 2014 Outercurve Foundation.

Discovering: System.Numerics.Vectors.Tests
Discovered:  System.Numerics.Vectors.Tests
Starting:    System.Numerics.Vectors.Tests

Assert failure(PID 16637 [0x000040fd], Thread: 16650 [0x410a]): Assertion failed '(op2 == idx) || (op2->gtEffectiveVal() == idx)' in 'System.Numerics.Vector:Narrow(struct,struct):struct' (IL size 84)

    File: /home/dragon/dotnet/arm32-dotnet/coreclr/src/jit/gentree.cpp Line: 3954
    Image: /home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10241504/overlay/dotnet


Assert failure(PID 16637 [0x000040fd], Thread: 16650 [0x410a]): Assertion failed '(op2 == idx) || (op2->gtEffectiveVal() == idx)' in 'System.Numerics.Vector:Narrow(struct,struct):struct' (IL size 84)

    File: /home/dragon/dotnet/arm32-dotnet/coreclr/src/jit/gentree.cpp Line: 3954
    Image: /home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10241504/overlay/dotnet

./RunTests.sh: line 86: 16637 Aborted                 (core dumped) $RUNTIME_PATH/dotnet xunit.console.netcore.exe System.Numerics.Vectors.Tests.dll -xml testResults.xml -notrait category=nonnetcoreapptests -notrait category=nonlinuxtests -notrait category=failing
~/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10241504/tests/System.Numerics.Vectors.Tests/netcoreapp-Linux-Release-x64
Finished running tests. End time=15:41:42. Return value was 134
Unable to find executable corerun
error: One or more tests failed while running tests from ''.  Exit code 134.
1 test(s) failed
```

```
 dragon@loki  ~/dotnet/arm32-dotnet/tmp  cat coreoverlay-ARM-CHE-REL-10241504/COMMITS.CLR 
2207dadf19fc976c6ccd04e12d46924a7b733457 Report registers as dead in GCInfo before the RhpPInvoke helper. (#14664)
5d66791a3a1932d2afcd3ef58dd0df91323662cb profiler changes for tiered compilation (#14612)
f8d6405c8e0680a1184dc07123aeb4abe62e9e87 Merge pull request #14647 from BruceForstall/AddArm64FrameLayoutDoc
67a66156d0307b49522a13618522d289268bc1f9 Fix for #12609 - add option to disable tiered compilation for profilers  (#14643)
33159233fd161877ffe37187bf06689fe0e8bea2 Fix RID extraction in packages and test build for Alpine (#14656)
458c90a363b22fb29f6581610d5349bf06e17971 Merge pull request #14649 from mikedn/float-absneg
2ba74317c027eb4ff1e854de02fbb9378d00a34e [Arm64] SIMD simple defines (#14628)
c0aa11ff835222facec5401c4074e5b46cb1635f [Arm64] SIMD lsra (#14631)
e9437e8a686d4db2a6667eb7f386279678916467 [Arm64] SIMD ins_Load (#14636)
4d253390b02d51dd5dc2049dbe83c122c3acba9d Merge pull request #14528 from jashook/pri1_rework

 dragon@loki  ~/dotnet/arm32-dotnet/tmp  cat coreoverlay-ARM-CHE-REL-10241504/COMMITS.FX 
7e91c485254e054e0f3c14291f97e6a382659bec Fix condition in System.Diagnostics.PerformanceCounter (#24828)
bcc59b2e74090cfbcc4d6e7c245c172744e9d46b Fix System.Data.Odbc configurations for support package and run UpdateVSConfigurations (#24752)
389d7ee0630e71ff868b4d9ebcdcf8111d5acee3 Merge pull request #24684 from Anipik/newbr
5d164eb843b6fc2278c107184634396642483901 Re-enabling previously disabled tests due to Retain bug (#24785)
1eabfe92ef9fe97f8abfccb044fd6bcfffa5bb5f Add some System.Collections.Concurrent perf tests (#24810)
a0c65af98df8fe746bb49da90b6c2a152d747a18 Remove New APIs in Syndication. (#24814)
d33e40e27e40d313977d324ae89c33f2842ee5b7 Deleted dead code from System.Net.Requests. (#24815)
c92789464b2ac983eba3a6021b13ef07eb0c8d8d Extentionclass added
725495577f7e17f1ae2ad2439f219203b2cef844 Add perf test to measure Memory.Span.Slice vs. Memory.Slice.Span (#24780)
90edf7d04a1d32266c275aa5b239b6022ebab98b Change Memory.Slice(...).Span to Memory.Span.Slice(...) (#24813)
```

**Comment by CarolEidt** [comment]
This was a problem with the assert in the case where we are spilling the odd half of a double (which has a float in it), and the assert was incorrect.
It was fixed by #14534
